### PR TITLE
flowctl: normalize --prefix trailing slash to avoid misleading PermissionDenied

### LINKED ISSUE #2834: flowctl catalog: missing trailing slash on --prefix gives misleading PermissionDenied error
When a user passes `--prefix <tenant>` without a trailing slash, the GraphQL API returns a `PermissionDenied` error even if the user has full admin access. The actual cause is that the `Prefix` type requires a trailing slash — the value is silently passed as-is to the API, which rejects it as an unrecognized/unauthorized prefix name.

**Steps to reproduce:**
```
flowctl catalog pull-specs --prefix AcmeCo
# Error: PermissionDenied: johnsmith@acme.co is not authorized to access prefix or name 'AcmeCo' with required capability read
```

**Expected:**
Either auto-normalize (`AcmeCo` → `AcmeCo/`) or return a clear error: `prefix must end with '/': did you mean 'AcmeCo/'?`

**Workaround:**
```
flowctl catalog pull-specs --prefix AcmeCo/
```

**Location:** `crates/flowctl/src/catalog/list/mod.rs` line ~272, where `models::Prefix::new(prefix)` is called without validating/normalizing the trailing slash. The same fix should apply to any other command using `NameSelector` with `--prefix`.

**Labels:** bug

### PR DESCRIPTION
**Pull request by GregorShear**
## Problem

`flowctl catalog pull-specs --prefix AcmeCo` (no trailing slash) returns a misleading error even for fully-authorized users:

```
PermissionDenied: johnsmith@acme.co is not authorized to access prefix or name 'AcmeCo' with required capability read
```

The value is passed verbatim to `models::Prefix::new`. Estuary prefixes must end in `/`, so the control plane treats `AcmeCo` as an unrecognized catalog *name* rather than a prefix and rejects it.

## Fix

Auto-normalize the prefix by appending a trailing `/` when it's missing. The fix lives in `to_vars` — the single point through which every `NameSelector`-based command routes its `--prefix` values:

- `catalog list`
- `catalog pull-specs`
- `catalog delete`

An **empty** prefix is deliberately left untouched: `""` is a valid selector meaning "everything the user can access", whereas `"/"` is invalid, so blindly appending would introduce a new bug.

## Testing

- New unit test `test_normalize_prefix` covers the missing-slash case, multi-segment prefixes, already-normalized input, and the empty edge case.
- `cargo test -p flowctl normalize_prefix` passes.

## Out of scope

`flowctl alerts subscriptions` and `flowctl raw` accept their own `--prefix` args that don't go through `NameSelector`, so they retain the same foot-gun. The issue scoped the fix to `NameSelector`; happy to fold those in as a follow-up if desired.

Closes #2834

### CODE REVIEW

File: crates/flowctl/src/catalog/list/mod.rs [hunk 1]
```diff
@@ -273,7 +273,7 @@ fn to_vars(list: &List) -> Vec<list_live_specs_query::LiveSpecsBy> {
     for prefix in list.name_selector.prefix.iter() {
         vars.push(list_live_specs_query::LiveSpecsBy {
             names: None,
-            prefix: Some(models::Prefix::new(prefix)),
+            prefix: Some(normalize_prefix(prefix)),
```
Comment (jshearer) [inline crates/flowctl/src/catalog/list/mod.rs]: Nit: Overlapping/duplicate prefixes that previously failed fast now normalize to the same value and produce duplicate rows. This is pre-existing, just slightly more confusing now if you do something unusual.

---

Ex:`flowctl catalog list --prefix acmeCo --prefix acmeCo/` (or the same bare prefix twice) 

Before this change the bare `acmeCo` format errored (`PermissionDenied`), so the command failed fast. Now, both entries become `acmeCo/`, `to_vars` pushes two identical `LiveSpecsBy` queries, and `fetch_paginated_live_specs` concatenates both result sets with no deduplication — every spec under `acmeCo/` is listed twice.

**Review by jshearer — DISMISSED** [review]
LGTM % one nit that was mostly pre-existing anyway

File: crates/flowctl/src/catalog/list/mod.rs (hunk 1 above)
Comment (GregorShear) [reply crates/flowctl/src/catalog/list/mod.rs]: fixed - deduping prefixes now

**Review by jshearer — APPROVED** [review]
LGTM

**Comment by telcharr** [comment]
For the record, #3035 which was opened June 15, linked to this same issue implemented this same normalization, including the empty-prefix handling and equivalent tests, plus the alerts subscriptions and raw coverage you've listed as "out-of-scope follow-up". Happy to rebase that remainder as the follow-up, but I would've appreciated a review or at least a heads-up on the original.
